Tests fail in before because of atomix supplier

This exception happens a lot in the last days, it happens in different tests.

```
16:02:24  [ERROR] shouldCompleteWorkflow(io.zeebe.broker.engine.TimerStartEventTest)  Time elapsed: 3.361 s  <<< FAILURE!
16:02:24  java.lang.AssertionError: 
16:02:24  
16:02:24  Expecting actual not to be null
16:02:24  	at io.zeebe.test.broker.protocol.clientapi.ClientApiRule.fetchAtomix(ClientApiRule.java:129)
16:02:24  	at io.zeebe.test.broker.protocol.clientapi.ClientApiRule.before(ClientApiRule.java:99)
16:02:24  	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:46)
16:02:24  	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
16:02:24  	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
16:02:24  	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
16:02:24  	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
16:02:24  	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
16:02:24  	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
16:02:24  	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
16:02:24  	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
16:02:24  	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
16:02:24  	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
16:02:24  	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
16:02:24  	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
16:02:24  	at org.junit.runners.Suite.runChild(Suite.java:128)
16:02:24  	at org.junit.runners.Suite.runChild(Suite.java:27)
16:02:24  	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
16:02:24  	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
16:02:24  	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
16:02:24  	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
16:02:24  	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
16:02:24  	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
16:02:24  	at org.apache.maven.surefire.junitcore.JUnitCore.run(JUnitCore.java:55)
16:02:24  	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:137)
16:02:24  	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeLazy(JUnitCoreWrapper.java:119)
16:02:24  	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:87)
16:02:24  	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75)
16:02:24  	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:158)
16:02:24  	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
16:02:24  	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
16:02:24  	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
16:02:24  	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
```

--- a/broker-core/src/test/java/io/zeebe/broker/test/EmbeddedBrokerRule.java
+++ b/broker-core/src/test/java/io/zeebe/broker/test/EmbeddedBrokerRule.java
@@ -312,9 +312,12 @@ public class EmbeddedBrokerRule extends ExternalResource {
       throw new RuntimeException(e);
     }
 
+    // capture value before removing the service as it will uninject the value on stop
+    final S value = injector.getValue();
+
     serviceContainer.removeService(accessorServiceName);
 
-    return injector.getValue();
+    return value;
   }
 
   public <T> void installService(

